enterprise/views.py

```diff
@@ -498,35 +499,68 @@ def set_final_prices(self, modes, request):
             result.append(mode)
         return result
 
-    def get_base_details(self, enterprise_uuid, course_run_id):
+    def get_base_details(self, request, enterprise_uuid, course_run_id):
         """
         Retrieve fundamental details used by both POST and GET versions of this view.
 
         Specifically, take an EnterpriseCustomer UUID and a course run ID, and transform those
         into an actual EnterpriseCustomer, a set of details about the course, and a list
         of the available course modes for that course run.
         """
+        enterprise_catalog_uuid = request.GET.get('catalog')
         try:
             enterprise_customer = get_enterprise_customer_or_404(enterprise_uuid)
-            course, course_run = \
-                CourseCatalogApiServiceClient(enterprise_customer.site).get_course_and_course_run(course_run_id)
         except ImproperlyConfigured:
             raise Http404
 
+        enterprise_catalog = None
+        if enterprise_catalog_uuid:
+            try:
+                enterprise_catalog = enterprise_customer.enterprise_customer_catalogs.get(
+                    uuid=enterprise_catalog_uuid
+                )
+            except EnterpriseCustomerCatalog.DoesNotExist:
+                add_generic_info_message_for_error(request)
+
+        if enterprise_catalog:
+            course, course_run = enterprise_catalog.get_course_and_course_run(course_run_id)
+        else:
+            try:
+                course, course_run = CourseCatalogApiServiceClient(
+                    enterprise_customer.site
+                ).get_course_and_course_run(course_run_id)
+            except ImproperlyConfigured:
+                raise Http404
+
         if not course or not course_run:
-            LOGGER.warning('Failed to fetch course "{course}" or course run "{course_run}" details'.format(
-                course=course, course_run=course_run
-            ))
+            LOGGER.warning(
+                'Failed to fetch course "{course}" or course run "{course_run}" details for '
+                'enterprise "{enterprise_uuid}"'.format(
+                    course=course, course_run=course_run, enterprise_uuid=enterprise_customer.uuid
+                )
+            )
             raise Http404
```

Comment:
Don't raise a 404, display the generic error message.

Reply:
This will be handled as part of [ENT-717] (https://openedx.atlassian.net/browse/ENT-717).